File: Zig/Practice/Pointers/even_nums.zig
const std = @import("std");
const print = std.debug.print;

pub fn main() void {
    const start_num: u8 = 0;
    const end_num: u8 = 200;

    var count: u8 = 0;
    var sum: u8 = 0;

    evenNums(start_num, end_num, &count, &sum);

    print("Count of even numbers between {} and {}: {}\n",.{start_num, end_num, count});
    print("Sum of even numbers between {} and {}: {}\n",.{start_num, end_num, sum});
}

fn evenNums(start: u8, end: u8, count: *u8, sum: *u8) void {
    var i: u8 = start;
    while( start <= end)
    {
        if (i % 2 == 0)
        {
            count.* += 1;  // Dereference the pointer to increment the count
            sum.* += i; // Dereference the pointer to add to the sum
        }
    }
}

File: COP3502C/Practice/even_nums.c
// Program to print the count of even numbers between 1–200. Also print their sum.

#include <stdio.h>

void even_nums(int start, int end, int *count, int *sum);

int main(void)
{
    int start_num, end_num;
    int count = 0, sum = 0;

    printf("Enter start number: \n");
    scanf("%d", &start_num);

    printf("Enter end number: \n");
    scanf("%d", &end_num);

    even_nums(start_num, end_num, &count, &sum);

    printf("Count of even numbers between %d and %d: %d\n", start_num, end_num, count);
    printf("Sum of even numbers between %d and %d: %d\n", start_num, end_num, sum);
    return 0;
}

void even_nums(int start, int end, int *count, int *sum)
{
    for (int i = start; i <= end; i++)
    {
        if (i % 2 == 0)
        {
            (*count)++;  // Dereference the pointer to increment the count
            (*sum) += i; // Dereference the pointer to add to the sum
        }
    }
}